core/src/main/java/inetsoft/report/composition/execution/AssetQueryCacheNormalizer.java

```diff
@@ -318,6 +318,33 @@ private boolean canRemoveAlias(String alias) {
          }
       }
 
+      // check if the alias is referenced in other assemblies
+      for(Assembly assembly : box.getWorksheet().getAssemblies()) {
+         if(!(assembly instanceof TableAssembly) || assembly == table) {
+            continue;
+         }
+
+         ColumnSelection otherColumns = ((TableAssembly) assembly).getColumnSelection();
+
+         for(int i = 0; i < otherColumns.getAttributeCount(); i++) {
+            ColumnRef col = (ColumnRef) otherColumns.getAttribute(i);
+
+            if(!col.isExpression()) {
+               continue;
+            }
+
+            Pattern pattern = Pattern.compile(table.getName() + "\\[[\"|']" + alias +
```

Comment:
Pattern compilation is somewhat expensive. Move this outside the loop because it can be reused.